### PR TITLE
Proper handle of wrong phpdoc tag case

### LINKED ISSUE #1322: PhpdocScalarFixer fails when param named in wrong case
```php
/**
 * @Param string
 */
```
PhpdocScalarFixer fails on above example, with
```
exception 'ErrorException' with message 'Undefined offset: 1' in /devel/src/github/Strate/PHP-CS-Fixer/Symfony/CS/Fixer/Symfony/PhpdocScalarFixer.php:104
Stack trace:
#0 /devel/src/github/Strate/PHP-CS-Fixer/Symfony/CS/Fixer/Symfony/PhpdocScalarFixer.php(104): {closure}(8, 'Undefined offse...', '/devel/src/gith...', 104, Array)
#1 /devel/src/github/Strate/PHP-CS-Fixer/Symfony/CS/Fixer/Symfony/PhpdocScalarFixer.php(83): Symfony\CS\Fixer\Symfony\PhpdocScalarFixer->fixScalars(Object(Symfony\CS\DocBlock\Line), 'param')
#2 /devel/src/github/Strate/PHP-CS-Fixer/Symfony/CS/Fixer.php(206): Symfony\CS\Fixer\Symfony\PhpdocScalarFixer->fix(Object(SplFileInfo), '<?php\nuse Logic...')
#3 /devel/src/github/Strate/PHP-CS-Fixer/Symfony/CS/Fixer.php(148): Symfony\CS\Fixer->fixFile(Object(SplFileInfo), Array, false, false, Object(Symfony\CS\FileCacheManager))
#4 /devel/src/github/Strate/PHP-CS-Fixer/Symfony/CS/Console/Command/FixCommand.php(371): Symfony\CS\Fixer->fix(Object(Symfony\CS\Config\Config), false, false)
#5 /devel/src/github/Strate/PHP-CS-Fixer/vendor/symfony/console/Command/Command.php(259): Symfony\CS\Console\Command\FixCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#6 /devel/src/github/Strate/PHP-CS-Fixer/vendor/symfony/console/Application.php(878): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#7 /devel/src/github/Strate/PHP-CS-Fixer/vendor/symfony/console/Application.php(195): Symfony\Component\Console\Application->doRunCommand(Object(Symfony\CS\Console\Command\FixCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#8 /devel/src/github/Strate/PHP-CS-Fixer/vendor/symfony/console/Application.php(126): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#9 /devel/src/github/Strate/PHP-CS-Fixer/php-cs-fixer(44): Symfony\Component\Console\Application->run()
#10 {main}
```

I suggest to write a `PhpdocParamNameCaseFixer` and run it before `PhpdocScalarFixer`. I am ready to work on PR, but I should know how acceptable my suggestion is.

### PR DESCRIPTION
**Pull request by Strate**
Fixes #1322

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
:-1:

**Comment by GrahamCampbell** [comment]
This is wrong.

**Comment by GrahamCampbell** [comment]
`@Param` is not the same as `@param` and we can't treat it as if it is.

**Comment by Strate** [comment]
@GrahamCampbell Right now php-cs-fixer fails on some of files. And this PR fixes exactly one type of fail. Right now `@Param` and `@param` are the same things, because `\Symfony\CS\DocBlock\Tag::__construct` performs `strtolower` on a tag name: https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/1.10/Symfony/CS/DocBlock/Tag.php#L51

So, let separate things: if we need to fix phpdoc tags case problem, we should do it in a separate PR. This PR only about `Undefined offset: 1` error.

**Comment by Strate** [comment]
Also, as you can see, this fixer does not touch tag name.

**Comment by GrahamCampbell** [comment]
> this fixer does not touch tag name.

But if makes fixes where it shouldn't.

**Comment by GrahamCampbell** [comment]
what ever follows `@Param` should NOT be altered by our fixer.

**Comment by Strate** [comment]
> what ever follows @Param should NOT be altered by our fixer.

hmm, this has a sense. I'll fix it.

File: Symfony/CS/Fixer/Symfony/PhpdocScalarFixer.php [hunk 1]
```diff
@@ -100,7 +100,7 @@ public function fix(\SplFileInfo $file, $content)
     private function fixScalars(Line $line, $tag)
     {
         $content = $line->getContent();
-        $tagSplit = preg_split('/\s*\@'.$tag.'\s*/', $content);
+        $tagSplit = preg_split('/\s*\@'.$tag.'\s*/i', $content);
```
Comment (keradus) [inline Symfony/CS/Fixer/Symfony/PhpdocScalarFixer.php]:
```diff
         $tagSplit = preg_split('/\s*\@'.$tag.'\s*/', $content);
+
+        if (!isset($tagSplit[1])) {
+            return;
+        }
```

File: Symfony/CS/Fixer/Symfony/PhpdocScalarFixer.php [hunk 2]
```diff
@@ -101,6 +101,9 @@ private function fixScalars(Line $line, $tag)
     {
         $content = $line->getContent();
         $tagSplit = preg_split('/\s*\@'.$tag.'\s*/', $content);
+        if (!isset($tagSplit[1])) {
```
Comment (GrahamCampbell) [inline Symfony/CS/Fixer/Symfony/PhpdocScalarFixer.php]: Please leave a blank line before and after this if statement.

Comment (keradus) [inline Symfony/CS/Fixer/Symfony/PhpdocScalarFixer.php]: please add empty line before and after control structure.

Comment (GrahamCampbell) [inline Symfony/CS/Fixer/Symfony/PhpdocScalarFixer.php]: Like @keradus did...

File: Symfony/CS/Tests/Fixer/Symfony/PhpdocScalarFixerTest.php [hunk 3]
```diff
@@ -256,4 +256,28 @@ public function testEmptyDocBlock()
 
         $this->makeTest($expected);
     }
+
+    public function testPhpdocHasAWrongTagNameCase()
```
Comment (keradus) [inline Symfony/CS/Tests/Fixer/Symfony/PhpdocScalarFixerTest.php]: what does that case tests?

File: Symfony/CS/Fixer/Symfony/PhpdocScalarFixer.php (hunk 2 above)
Comment (GrahamCampbell) [inline Symfony/CS/Fixer/Symfony/PhpdocScalarFixer.php]: lol, we both commented at the same time :)

File: Symfony/CS/Tests/Fixer/Symfony/PhpdocScalarFixerTest.php (hunk 3 above)
Comment (Strate) [inline Symfony/CS/Tests/Fixer/Symfony/PhpdocScalarFixerTest.php]: Without fix this test fails with exception.

File: Symfony/CS/Fixer/Symfony/PhpdocScalarFixer.php (hunk 2 above)
Comment (Strate) [inline Symfony/CS/Fixer/Symfony/PhpdocScalarFixer.php]: Seems that we should have a fixer for this case :)

File: Symfony/CS/Tests/Fixer/Symfony/PhpdocScalarFixerTest.php (hunk 3 above)
Comment (keradus) [inline Symfony/CS/Tests/Fixer/Symfony/PhpdocScalarFixerTest.php]: as well as the second one, and this test name doesn't tell us what we are testing, the second test is better and may be the only one added test.

**Comment by keradus** [comment]
:+1:

File: Symfony/CS/Tests/Fixer/Symfony/PhpdocScalarFixerTest.php [hunk 4]
```diff
@@ -256,4 +256,16 @@ public function testEmptyDocBlock()
 
         $this->makeTest($expected);
     }
+
+    public function testWrongCasedPhpdocTagDoesNotAltered()
```
Comment (GrahamCampbell) [inline Symfony/CS/Tests/Fixer/Symfony/PhpdocScalarFixerTest.php]: The grammar is wrong here. Should be `IsNot` rather than `DoesNot`.

**Comment by GrahamCampbell** [comment]
Actually this PR looks wrong. :-1: for merging.

**Comment by GrahamCampbell** [comment]
Surely we need to fix this part of the code:

```php
$annotations = $doc->getAnnotationsOfType(self::$tags);
```

Incorrect annotations are being returned from here.

**Comment by GrahamCampbell** [comment]
I've provided an alternative here: https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1327.

**Comment by GrahamCampbell** [comment]
Ping @keradus. I hope my PR fixes the underlying issue.

**Comment by keradus** [comment]
Closing due to merge of #1327.
Thank you @Strate

**Pull request closed without merging**